### PR TITLE
Marginal tax rate

### PR DESCRIPTION
**Pull request by clallemand**
#### Breaking changes

- In scenario/_apply_modification:
  - Deal with integer and small values in varying_variable

### CODE REVIEW

File: openfisca_survey_manager/scenarios.py [hunk 1]
```diff
@@ -1433,6 +1435,7 @@ def _apply_modification(self, simulation, period):
 
         def set_variable(varying_variable, varying_variable_value, period_):
             delta = self.variation_factor * varying_variable_value
+            delta = np.where(varying_variable_value == 0.0 ,1 , np.sign(delta)) * max_(100.0, abs_(delta))
```
Comment (benjello) [inline openfisca_survey_manager/scenarios.py:1438]: We should introduce an attribute to the scenario class to control for the minimal value which is arbitrary set to 100 here.

**Review by benjello — CHANGES_REQUESTED** [review]

**Review by nikhilwoodruff — COMMENTED** [review]
Thanks very much @clallemand and @benjello for this fix, it looks like it'll help with using it with OpenFisca-UK. I'm still curious about the effects of increasing the incomes of all members of a group at the same time: in our [original implementation](https://github.com/PSLmodels/openfisca-uk/blob/713c1286b21f09003deaaa2f41c1dfbb34ac4c34/openfisca_uk/tools/simulation.py#L361) we handled this by having each member of a group essentially take turns having their income increased and marginal tax rate calculated; it took a lot longer but I'm seeing some features of openfisca-core that might speed it up, e.g. the ```simulation.populations[group].members_position``` gives the index of each group member in their group, perhaps if there was a ```modified_simulation``` for each distinct value (UK data has at most 8 members of a family), and the marginal tax rate arrays could be combined using the same split. I'm happy to have a go at adding this if this would be beneficial?

**Comment by benjello** [comment]
@nikhilwoodruff : please do. As a suggestion you could add an dict attribute to the Scenario class specifying all the modified_simulation config.

**Comment by nikhilwoodruff** [comment]
@benjello Ok, will have a think about perhaps using a dict attribute. I had a [quick go](https://github.com/nikhilwoodruff/openfisca-survey-manager) at this yesterday and I think the functionality is there (and behaviour the same by default):

- ```mtr_group``` is an added optional attribute which if given preserves independence between members in the same group entity specified. E.g. if mtr_group = "household", the marginal tax rates will be calculated independently for each member of the household. 
- ```mtr_varying_variable_level``` is a boolean attribute I thought might be useful. When True, then the target variable values are projected onto the entity of the varying variable (e.g. if varying = "earnings" and target = "household_net_income", then we'll get an MTR for each person, rather than each household). When False, the values are averaged over the entity. This is slightly different to the previous approach (add the deltas, add the change in target vars, MTR = sum_change / sum_delta): instead this averages the two individual MTRs if a household-level MTR is needed. I *think* these two are largely the same, but I'm not 100% sure.

The group independence does multiply the time required (by the maximum group size), but it would be useful I think for our analyses. There's definitely some undefined behaviour for when the varying variables are group-level and other cases so it's still in progress, but I'm curious what you think of this functionality?

**Comment by benjello** [comment]
@clallemand : is this still up to be merged ?

**Pull request closed without merging**